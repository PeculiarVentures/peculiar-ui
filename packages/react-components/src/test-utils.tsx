import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
