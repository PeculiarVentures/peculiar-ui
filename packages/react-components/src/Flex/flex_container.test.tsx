import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { FlexContainer } from './index';

describe('<FlexContainer />', () => {
  it('should render with default styles', () => {
    const {
      asFragment,
    } = render(<FlexContainer>1</FlexContainer>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const {
      asFragment,
    } = render(<FlexContainer className="test-classname" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass gap', () => {
    const {
      asFragment,
    } = render(<FlexContainer gap={10} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass direction', () => {
    const {
      asFragment,
    } = render(<FlexContainer direction="column" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass wrap', () => {
    const {
      asFragment,
    } = render(<FlexContainer wrap="nowrap" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass align', () => {
    const {
      asFragment,
    } = render(<FlexContainer align="center" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass justify', () => {
    const {
      asFragment,
    } = render(<FlexContainer justify="flex-end" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass component', () => {
    const {
      asFragment,
    } = render(<FlexContainer component="li" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
