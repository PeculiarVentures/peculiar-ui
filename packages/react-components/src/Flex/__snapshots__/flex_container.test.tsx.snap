// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlexContainer /> should pass align 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`<FlexContainer /> should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
    class="test-classname emotion-0"
  />
</DocumentFragment>
`;

exports[`<FlexContainer /> should pass component 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<li
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`<FlexContainer /> should pass direction 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`<FlexContainer /> should pass gap 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 10px;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`<FlexContainer /> should pass justify 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`<FlexContainer /> should pass wrap 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`<FlexContainer /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
    class="emotion-0"
  >
    1
  </div>
</DocumentFragment>
`;
